cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -5562,3 +5563,13 @@ if(ARROW_WITH_AZURE_SDK)
 endif()
 
 message(STATUS "All bundled static libraries: ${ARROW_BUNDLED_STATIC_LIBS}")
+
+# ----------------------------------------------------------------------
+# Apache Flight SQL ODBC
+
+if(ARROW_FLIGHT_SQL_ODBC)
+  find_package(ODBC REQUIRED)
+  if(MSVC)
+    find_package(Boost REQUIRED COMPONENTS locale)
+  endif()
+endif()
```

Comment:
Could you move this before `message(STATUS "All bundled static libraries: ${ARROW_BUNDLED_STATIC_LIBS}")`?